### ISSUE TITLE
Windows Insider: Unity editor hangs in play mode when using UnityEngine.VR.WSA.Input.GestureRecognizer

### ISSUE THREAD

**Issue by markneville** [Microsoft/MixedRealityToolkit-Unity#464]
Latest build version of InputManager.prefab causes Unity to hang.

Unity 5.50f3
Windows 10 

Cheers,
Mark

**Comment by HodgsonSDAS** [comment]
I mark, can you provide us with a bit more details?

Were you able to pinpoint the issue with a debugger?

Do you have any reproduction steps?

**Comment by NeerajW** [comment]
Looks like I could repro this with:
1. Import latest release asset package into Unity 5.5.0f3 project
2. Use the HoloLensCamera prefab
3. Create Managers object and child a InputManager prefab to it
4. Click Play in editor and observe hang. 

Working on more details.

**Comment by NeerajW** [comment]
I remember this not being an issue when I checked in InputTapTest.unity scene. I am reinstalling Unity to see if it helps.

**Comment by NeerajW** [comment]
@markneville are you on a Windows Insider Preview build? I could repro this issue only when I upgraded to a selfhost build.

**Comment by NeerajW** [comment]
To unblock yourself in the editor consider disabling the GestureInput component under InputManager just when in the editor.

**Comment by markneville** [comment]
@NeerajW Yes i'm on the Fast ring for the Windows Insider Preview build. Also the way you repro'ed the issue was exactly the same way I found the issue. I'll try disabling the GestureInput component to see if that helps.

**Comment by NeerajW** [comment]
@markneville thanks for confirming. I have filed a Windows OS bug to track this issue and see where the real problem is.

**Comment by 3cst4sy** [comment]
Hi, i run into the same issue. Disabling GestureInput helps to start the playmode, but that's not a fix.
Unity: 5.5.2f1 HoloToolkit 1.5.6
Any further informations?

**Comment by ujell** [comment]
I started to have this problem since I reinstalled Windows 10 (version 1703). It seems like Windows or Unity bug since I can reproduce it without HoloToolkit; creating a new GestureRecognizer is freezing the editor even in an empty project (in HoloToolkit [this line](https://github.com/Microsoft/HoloToolkit-Unity/blob/master/Assets/HoloToolkit/Input/Scripts/InputSources/GesturesInput.cs#L34)). I tried Unity 5.5.3p2 and 5.6, both had the issue and Unity log doesn't have any related information.

In addition, someone in Unity forums also encountered [this](https://forum.unity3d.com/threads/unity-freezes-when-creating-a-new-gesturerecognizer-5-5-and-up.468514/).

**Comment by StephenHodgson** [comment]
Any update on this?

**Comment by ujell** [comment]
I had to clean install Windows (directly from USB instead of "Reset Windows" thingy) because of another issue and this was fixed afterwards by itself 🤷 . Someone on Unity forum said it's because of missing Windows 10 [SDK](https://forum.unity3d.com/threads/unity-freezes-when-creating-a-new-gesturerecognizer-5-5-and-up.468514/#post-3082081), but I'm pretty sure everything was installed correctly when I was having this issue.

**Comment by jessemcculloch** [comment]
I think if nobody comments that this is still an issue, we can probably close this one

**Comment by michael-house** [comment]
I'm seeing this in Unity 2017.1.1f1. Having `GesturesInputSource` active in the scene causes hang when playing the scene. No error output in the log file. I attached the debugger and couldn't step past the first line in the `Start()` method: `gestureRecognizer = new GestureRecognizer();`. 

Oddly enough, this project was functional on the same computer earlier in the day. After a reboot from some Windows Updates, this problem presented. 
![image](https://user-images.githubusercontent.com/29042601/31370838-a52276ac-ad42-11e7-8b57-97e18422b85a.png)

**Comment by michael-house** [comment]
After multiple restarts, reinstalling Unity and reverting source back to a known good state, the problem still persisted.

I enabled Developer Mode in Windows and restarted, now the problem is gone.
